system/lib/emmalloc.c

```diff
@@ -791,6 +789,13 @@ static void *allocate_memory(size_t alignment, size_t size)
 
   // We were unable to find a free memory region. Must sbrk() in more memory!
   size_t numBytesToClaim = size+sizeof(Region)*3;
+  // Take into account the alignment as well. For typical alignment we don't
+  // need to add anything here (so we do nothing if the alignment is equal to
+  // MALLOC_ALIGNMENT), but it can matter if the alignment is very high. In that
+  // case, not adding the alignment can lead to this sbrk not giving us enough
+  // (in which case, the next attempt fails and will sbrk the same amount again,
+  // potentially allocating a lot more memory than necessary).
+  numBytesToClaim += alignment - MALLOC_ALIGNMENT;
```

Comment:
So `alignment - MALLOC_ALIGNMENT` is the most we would ever need to add to the resulting allocation to achieve and an alignment of `alignment`?

Reply:
I think so, yes. What I am not certain of is whether the `3*` figure a few lines above this include slack or if they are precise, which could matter here.

Another way to go could if `if alignment > malloc alignment then add the alignment`?

Reply:
I think doing it that way would make the code more straight forwardly correct to a reader.